src/app/dim-api/import.ts

```diff
@@ -149,16 +150,20 @@ export function importDataBackup(data: ExportResponse, silent = false): ThunkRes
 }
 
 /** Returns a promise that resolves when the profile is fully loaded. */
-function waitForProfileLoad() {
+function waitForProfileLoad<D extends Dispatch>(dispatch: D) {
+  const observerId = 'profile-load-observer';
```

Comment:
This fixed ID does mean that you can never have two of these observers at once. It probably doesn't mean much here, but it'd be better to have a unique ID given the contract of this function.

Reply:
Yeah the assumption was that it was only really run once on load and didn't matter, but I get it, will change.

Reply:
Just to reiterate, this doesn't run on load - it only runs when a user imports a data backup manually

Reply:
Yes, understood